automata/fa/fa.py

```diff
@@ -1,19 +1,230 @@
 #!/usr/bin/env python3
 """Classes and methods for working with all finite automata."""
+from __future__ import annotations
 
 import abc
-from typing import Set
+import os
+import pathlib
+import uuid
+from collections import defaultdict
+from typing import Dict, Generator, List, Literal, Optional, Set, Tuple, Union
 
 from automata.base.automaton import Automaton, AutomatonStateT
 
+# Optional imports for use with visual functionality
+try:
+    import coloraide
+    import pygraphviz as pgv
+except ImportError:
+    _visual_imports = False
+else:
+    _visual_imports = True
+
+
 FAStateT = AutomatonStateT
+LayoutMethod = Literal["neato", "dot", "twopi", "circo", "fdp", "nop"]
 
 
 class FA(Automaton, metaclass=abc.ABCMeta):
     """An abstract base class for finite automata."""
 
     __slots__ = tuple()
 
+    @staticmethod
+    def get_state_name(state_data: FAStateT) -> str:
+        """
+        Get an string representation of a state. This is used for displaying and
+        uses `str` for any unsupported python data types.
+        """
+        if isinstance(state_data, str):
+            if state_data == "":
+                return "λ"
```

Comment:
@eliotwrobson @khoda81 What is the difference between `𝛌` or `ε` in an Automaton Theory context? I've seen both used, sometimes seemingly interchangeably, to represent empty string.

Reply:
There is no difference. It just depends on which symbol you want to use to represent the empty; I prefer `ε`, though we should probably stick to one convention in this library.

Reply:
@eliotwrobson That's what I thought, thanks. The class I took in university used "lambda" almost ecxlusively—I didn't learn about "epsilon" as an alternative term until I googled more about the automata theory (after I graduated).

A quick search through the codebase's Markdown files reveals that we seem to lack a lean towards one or the other term (i.e. "lambda" vs. "epsilon").

<img width="572" alt="Screenshot 2023-05-18 at 9 04 09 PM" src="https://github.com/caleb531/automata/assets/872474/265cb883-487d-498e-b3e1-405f1be5e931">

<img width="399" alt="Screenshot 2023-05-18 at 9 04 01 PM" src="https://github.com/caleb531/automata/assets/872474/0e1b71cf-c669-45e7-8cf6-cc69f2d866ad">

I suppose you could argue that the name of our existing method `eliminate_lambda` implies that 'lambda' is the convention, but I don't think that's enough to establish a preference. I think that's why I refer to both terms in the documentation often, so that people who are familiar with either convention

I _was_ debating if this would be worthy of being a function parameter or even a global config option, so that users could choose one or the other.

Reply:
Going to go ahead and make this into a separate issue (and resolve) since it's somewhat orthogonal to the goals of this PR, and shouldn't really be THAT much work.